paddle/fluid/operators/mkldnn/layer_norm_mkldnn_op.cc

```diff
@@ -0,0 +1,177 @@
+/* Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/fluid/operators/layer_norm_op.h"
+#include "paddle/fluid/platform/mkldnn_reuse.h"
+
+namespace paddle {
+namespace operators {
+
+template <typename T>
+class LayerNormMKLDNNHandler
+    : public platform::MKLDNNHandlerT<T, dnnl::layer_normalization_forward> {
+ public:
+  LayerNormMKLDNNHandler(const std::vector<int64_t>& dims, const float& epsilon,
+                         const dnnl::normalization_flags& flags,
+                         const bool& is_test, const MKLDNNMemoryFormat fmt,
+                         const platform::MKLDNNDeviceContext& dev_ctx,
+                         platform::Place cpu_place,
+                         const std::string& uniq_name)
+      : platform::MKLDNNHandlerT<T, dnnl::layer_normalization_forward>(
+            dev_ctx, dev_ctx.GetEngine(), cpu_place,
+            platform::CreateKey(dims, uniq_name)) {
+    if (!this->isCached()) {
+      auto md = dnnl::memory::desc(dims, platform::MKLDNNGetDataType<T>(), fmt);
+      if (!is_test) {
+        // TODO(grygielski) Delete forcing stats_md after DNNL 1.2 is introduced
```

Comment:
I talked with @grygielski and this may require more changes and it is only relevant to training (inference does not go there) so I added an issue to our system to analyze this problem later on.